test/integration/other/analytics-queries.js

```diff
@@ -1343,6 +1344,507 @@ describe('analytics task queries', function () {
+          .replace('two', 'two-update')
+          .replace('baseVersion=""', 'baseVersion="1"')
+          .replace('<status>new</status>', '<status>checked in</status>')
+        )
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await exhaust(container);
+
+      backlogCount = await container.oneFirst(sql`select count(*) from entity_submission_backlog`);
+      backlogCount.should.equal(1);
+
+      // Send the second submission to create the entity, which should trigger
+      // the processing of the next submission in the branch.
+      await asAlice.post('/v1/projects/1/forms/offlineEntity/submissions')
+        .send(testData.instances.offlineEntity.two)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await exhaust(container);
+
+      // Two reprocessing events logged now
+      countReprocess = await container.Analytics.countSubmissionReprocess();
+      countReprocess.should.equal(2);
+    }));
+
+    it('should measure time from submission creation to entity version finished processing', testService(async (service, container) => {
+      await createTestForm(service, container, testData.forms.offlineEntity, 1);
+
+      const asAlice = await service.login('alice');
+
+      // Make an entity that doesn't have a source
+      await asAlice.post('/v1/projects/1/datasets/people/entities')
+        .send({
+          uuid: '12345678-1234-4123-8234-123456789abc',
+          label: 'Johnny Doe',
+          data: { first_name: 'Johnny', age: '22' }
+        })
+        .expect(200);
+
+      // times may be null if no submissions have been processed
+      let waitTime = await container.Analytics.measureEntityProcessingTime();
+      waitTime.should.eql({ max_wait: null, avg_wait: null });
+
+      // Make an entity update
+      await asAlice.post('/v1/projects/1/forms/offlineEntity/submissions')
+        .send(testData.instances.offlineEntity.one
+          .replace('branchId=""', `branchId="${uuid()}"`)
+        )
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      // Make another entity create
+      await asAlice.post('/v1/projects/1/forms/offlineEntity/submissions')
+        .send(testData.instances.offlineEntity.two)
+        .set('Content-Type', 'application/xml')
+        .expect(200);
+
+      await exhaust(container);
+      waitTime = await container.Analytics.measureEntityProcessingTime();
+      waitTime.max_wait.should.be.greaterThan(0);
+      waitTime.avg_wait.should.be.greaterThan(0);
+    }));
+
+    it('should not see a delay for submissions processed with approvalRequired flag toggled', testService(async (service, container) => {
+      const asAlice = await service.login('alice');
+
+      // Create form, set dataset people to approvalRequired = true
+      await createTestForm(service, container, testData.forms.simpleEntity, 1);
+      await approvalRequired(service, 1, 'people');
```

Comment:
Ah sorry, I just removed this function in #1196, as it wasn't used anymore and the linter was complaining. I think merging now would break the `master` branch, but you could merge `master` into this PR, then add `approvalRequired()` back. Or if it's not used elsewhere, you could just send the request to toggle the flag without using `approvalRequired()`. Sorry about that!